### PR TITLE
Commander to yargs

### PR DESCRIPTION
**Pull request by szeck87**
This is initial work to use yargs instead of commander.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,17 +46,24 @@
...
+    "ts-jest": "^23.0.0",
     "tslint": "5.10.0",
     "typedoc": "0.11.1",
     "typescript": "2.9.2"
   },
   "dependencies": {
-    "commander": "^2.9.0",
+    "@types/node": "^10.5.1",
+    "chalk": "^2.4.1",
+    "commander": "^2.15.1",
```
Comment (Glavin001) [inline package.json]: Should remove `commander`.

**Comment by szeck87** [comment]
@Glavin001 can you look into how the tests run?  I'm guessing it doesn't like parts of `Runner` being async now.  Not sure how to fix it.

File: test/Runner.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { Runner } from "../src/Runner";
+
+describe("Runner", () => {
+  class CustomRunner extends Runner {
+    public stdout: string = "";
+    public stderr: string = "";
+    public exitCode: number;
```
Comment (Glavin001) [inline test/Runner.spec.ts]: I think `exitCode` should default to `0`. Silly mistake on my part. Appreciate if you could fix :).

File: test/Runner.spec.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        stderr: this.stderr,
+        stdout: this.stdout,
+      };
+    }
+  }
+  describe("Support", () => {
+    describe("with JSON", () => {
+      test("languages", () => {
+        const runner = new CustomRunner();
+        runner.support({
```
Comment (Glavin001) [inline test/Runner.spec.ts]: This is async now, right? You may need to `await`.
And make sure to either return a Promise or use some other method to let Jest know this test is async.

File: test/Runner.spec.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          json: true,
+          languages: true,
+        });
+        expect(runner.toJSON()).toMatchSnapshot();
+      });
+    });
+    describe("without JSON", () => {
+      test("languages", () => {
+        const runner = new CustomRunner();
+        runner.support({
```
Comment (Glavin001) [inline test/Runner.spec.ts]: Same as above. Should handle async within test.

File: test/Runner.spec.ts (hunk 3 above)
Comment (szeck87) [reply test/Runner.spec.ts]: I needed `async` in both the `describe` and `test` blocks.  I had it only in the `test` block before.

**Comment by szeck87** [comment]
Need to go through the interfaces in Runner.ts and make any changes necessary now that it's using yargs.

**Comment by szeck87** [comment]
@Glavin001 can you play around with this and approve if you're good with these changes?

**Comment by Glavin001** [comment]
@szeck87 I'll take it from here 👍

**Pull request merged**